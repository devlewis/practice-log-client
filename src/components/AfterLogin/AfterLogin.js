import React, { PureComponent } from "react";
import "./AfterLogin.css";
import Context from "../../Context";
import TokenService from "../../services/token-service";
import IdleService from "../../services/idle-service";

class AfterLogin extends PureComponent {
  static contextType = Context;

  handleLoginFetch = () => {
    this.context.onHandleLoginFetch(this.props.history);
  };

  handleLogout = () => {
    TokenService.clearAuthToken();
    /* when logging out, clear the callbacks to the refresh api and idle auto logout */
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <button onClick={() => this.handleLoginFetch()}>
          Take me to my current goal!
        </button>

        <button onClick={() => this.props.history.push("/setup")}>
          Set up a new goal!
        </button>
        <button onClick={this.handleLogout}>Logout</button>
      </div>
    );
  }
}

export default AfterLogin;
