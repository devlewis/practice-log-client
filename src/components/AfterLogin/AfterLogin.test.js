import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AfterLogin from "./AfterLogin";

describe(`AfterLogin component`, () => {
  const props = {
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop",
  };

  it("renders form.AfterLogin by default", () => {
    const wrapper = shallow(<AfterLogin />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the AfterLogin given props", () => {
    const wrapper = shallow(<AfterLogin {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
