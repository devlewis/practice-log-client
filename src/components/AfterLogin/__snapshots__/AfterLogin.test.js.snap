// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AfterLogin component renders form.AfterLogin by default 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Take me to my current goal!
  </button>
  <button
    onClick={[Function]}
  >
    Set up a new goal!
  </button>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;

exports[`AfterLogin component renders the AfterLogin given props 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Take me to my current goal!
  </button>
  <button
    onClick={[Function]}
  >
    Set up a new goal!
  </button>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
